src/main/java/org/sagebionetworks/web/client/widget/provenance/ProvViewUtil.java

```diff
@@ -78,19 +81,26 @@ public static ProvNodeContainer createActivityContainer(ActivityGraphNode node,
 				label.add(new HTML(DisplayUtils.stubStrPartialWord(node.getActivityName(), MAX_ACT_CODE_NAME_CHAR)));
 			}
 			UserBadge badge = ginInjector.getUserBadgeWidget();
-			badge.configure(node.getModifiedBy());
 			HTML time = new HTML(ginInjector.getDateTimeUtils().getLongFriendlyDate(node.getModifiedOn()));
 			time.addStyleName(PROV_ACTIVITY_TIME_STYLE);
-
 			FlowPanel content = new FlowPanel();
 			label.addStyleName(ACT_MARGIN_NAME);
 			content.add(label);
-			Widget userBadgeWidget = badge.asWidget();
-			userBadgeWidget.addStyleName(ACT_MARGIN_USER);
-			content.add(userBadgeWidget);
+			Div badgeContainer = new Div();
+			badgeContainer.addStyleName(ACT_MARGIN_USER);
+			badgeContainer.add(badge.asWidget());
+			content.add(badgeContainer);
 			time.addStyleName(ACT_MARGIN_TIME);
+			if (DisplayUtils.isDefined(node.getActivityDescription())) {
+				Paragraph descriptionParagraph = new Paragraph(DisplayUtils.stubStrPartialWord(node.getActivityDescription(), MAX_ACT_CODE_NAME_CHAR));
+				descriptionParagraph.addStyleName(PROV_ACTIVITY_LABEL_STYLE);
+				descriptionParagraph.addStyleName(ACT_MARGIN_TIME);
+				Tooltip description = new Tooltip(descriptionParagraph, node.getActivityDescription());
+				content.add(description);
+			}
 			content.add(time);
 			container.addContent(content);
+			badge.configure(node.getModifiedBy());
```

Comment:
Do not configure the UserBadge until after it's been attached to the parent (was not rendering).